ccswbs/gus
# New widget: UoG events teaser display

# Summary of changes
Add a new widget to show upcoming UoG events. These events are pulled from the UoG News WordPress site. Also add new feed and taxonomy for UoG event categories to the Drupal backend.

## Frontend
- Add `gatsby-source-wordpress` plugin
- Update `gatsby-node.js` with schema for both WordPress events and the new Events widget
- Add Events graphQL fragments to `widgets.js`
- Add `events.js` component
- General code cleanup

## Backend
- Add Events Widget paragraph type
- Make Events Widget available to Basic Page widget selection
- Rearrange Basic Page widget buttons in alphabetical order
- Add [Feeds extensible parsers](https://www.drupal.org/project/feeds_ex) module
- Add [JsonPath library](https://github.com/Peekmo/JsonPath)
- Add `Event Categories import` feed type
- Add `Event Categories` feed
- Add `Event Categories` Taxonomy vocabulary

# Test Plan

Gatsby Cloud URL: https://tqtest.gatsbyjs.io/
Drupal multidev: https://ugevents-chug.pantheonsite.io/

1. Go to https://tqtest.gatsbyjs.io/widget-options
2. Verify the first Events Widget shows events with category `Homepage`
3. Verify the second Events Widget shows events with both the categories `Homepage` and `Intranet Events`
4. Verify the third Events Widget shows events that only have the category `Intranet Events`
5. Create a new basic page on https://ugevents-chug.pantheonsite.io/ and add Event widgets. If you leave `Match all categories` unchecked, the teaser will display events that have any one of your selected categories. If you check `Match all categories` it will only display events whose categories are an exact match 

**Note:** to see the categories on the original WordPress events, either run a local build using https://ugevents-chug.pantheonsite.io/ as base url and explore the GraphQL schema, or else go here - https://news.uoguelph.ca/events/ - and click through to the event in question

**Known issues:** The raw logs on Gatsby Cloud show warnings about `gatsby-source-wordpress` being incompatible with Gatsby versions older than 4, but this does not appear to affect the plugin's functionality.
